src/api/app/controllers/person/token_controller.rb

```diff
@@ -0,0 +1,42 @@
+require 'xmlhash'
+module Person
+  class TokenController < ApplicationController
+    before_action :set_user, only: [:index, :create, :delete]
+
+    # GET /person/<login>/token
+    def index
+      authorize @user, :show?
+
+      @list = @user.service_tokens
+    end
+
+    # POST /person/<login>/token
+    def create
+      authorize @user, :update?
+
+      pkg = nil
+      if params[:project] || params[:package]
```

Comment:
What if there is `:project` but not `:package`?

Reply:
I think it's a bug, but I didn't want to change it altogether. I also have a commit which handles invalid parameters, but want to have this merged first.

Reply:
k